sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggregationIterator.scala

```diff
@@ -181,11 +182,7 @@ abstract class AggregationIterator(
         // Process all expression-based aggregate functions.
         updateProjection.target(currentBuffer)(joinedRow(currentBuffer, row))
         // Process all imperative aggregate functions.
-        var i = 0
-        while (i < updateFunctions.length) {
-          updateFunctions(i)(currentBuffer, row)
-          i += 1
-        }
+        updateFunctions.foreach(updateFunction => updateFunction(currentBuffer, row))
```

Comment:
I just wonder if it'd make sense if we cache the length and use just a normal `while` loop as done above. This seems not encouraged [1]. I believe this is fine but not for critical path. IIUC, this will , at least, introduce virtual function calls for each row. I found a blog just to be verbose to make sure myself [2]. I am worried because this change can be a example in the future. Maybe, we need a clear reason to change this.

[1] https://github.com/databricks/scala-style-guide#traversal-and-zipwithindex
[2] http://www.scalablescala.com/roller/scala/entry/why_is_using_for_foreach